lib/validator/common.go

```diff
@@ -80,3 +82,39 @@ func ValidateMetadataIDsAssigned(rm unversioned.ResourceMetadata) error {
 
 	return nil
 }
+
+// ValidateIDFieldsSame is used to validate that the Resource Objects
```

Comment:
I think this is only useful for calicoctl, so I'd not include this in libcalico-go.